### ISSUE TITLE
In-situ processing of joinattributestable

### ISSUE THREAD

**Issue by Martin-BKT** [qgis/QGIS#50136]
### Feature description

With joinattributestable processing, data from a second layer can be added to a layer based on common identifiers, quite like "join" in SQL databases. However, it is not possible to add the information to the layer itself, but only generate a new layer with the joined information, since there is no in-situ mode for joinattributestable. It would be great if in-situ mode could be added, so the base layer can remain as is, with the information from the other layer added.

**Comment by Martin-BKT** [comment]
> Why not use the [joins properties](https://docs.qgis.org/testing/en/docs/user_manual/working_with_vector/vector_properties.html#joins-properties) feature of the layer?

Joins properties apparently does the job. However, this only works within the current QGIS project since the join is stored in the project, not in the layer itself. The current workaround is to work with joins-properties and then copy each field individually using the field calculator, and remove the join after this is done. It would simply be nice if that could be avoided.

Another issue with using joins properties is the fact that a change in the related table will also change the data in the current table. joinattributestable takes the current snapshot of the data in the related layer.

**Comment by DelazJ** [comment]
> The current workaround is to work with joins-properties and then copy each field individually using the field calculator, and remove the join after this is done. It would simply be nice if that could be avoided.

Right-click > Export > Save As

**Comment by Martin-BKT** [comment]
> Right-click > Export > Save As

Yes, of course this is the second workaround. But then I have a new table instead of an updated existing table. Yes, I can delete the old one, replace it with the new one, and copy the style.

However, that could be done with all the processing tools. Some of them provide in-situ processing which makes life a lot easier. It would just be so great if the workaround could be avoided, just like with other processing tools that do provide in-situ processing.
